src/Form/InvoiceSetPaymentDateForm.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+/*
+ * This file is part of the Kimai time-tracking app.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace App\Form;
+
+use App\Entity\Customer;
+use App\Entity\Invoice;
+use App\Entity\Project;
+use App\Form\Type\CustomerType;
+use App\Form\Type\DateTimePickerType;
+use App\Repository\CustomerRepository;
+use App\Repository\Query\CustomerFormTypeQuery;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\Extension\Core\Type\TextareaType;
+use Symfony\Component\Form\Extension\Core\Type\TextType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+class InvoiceSetPaymentDateForm extends AbstractType
+{
+    use EntityFormTrait;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $dateTimeOptions = [
+            'model_timezone' => $options['timezone'],
+            'view_timezone' => $options['timezone'],
+        ];
+        // primarily for API usage, where we cannot use a user/locale specific format
+        if (null !== $options['date_format']) {
+            $dateTimeOptions['format'] = $options['date_format'];
+        }
+
+        $builder
+            ->add('paymentDate', DateTimePickerType::class, array_merge($dateTimeOptions, [
+                'label' => 'label.paymentDate',
+                'required' => true,
+            ]));
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function configureOptions(OptionsResolver $resolver)
+    {
+        $resolver->setDefaults([
+            'data_class' => Invoice::class,
+            'csrf_protection' => true,
+            'csrf_field_name' => '_token',
+            'csrf_token_id' => 'admin_invoice_set_payment_date',
+            'currency' => Customer::DEFAULT_CURRENCY,
+            'date_format' => null,
+            'include_budget' => false,
+            'timezone' => date_default_timezone_get(),
+            'time_increment' => 1,
```

Comment:
I'm not sure if I need the option `time_increment` anyway.

Reply:
skip it, totally fine to set it to 30 fixed

Reply:
When I remove this option then I get `An error has occurred resolving the options of the form "App\Form\InvoicePaymentDateForm": The option "time_increment" does not exist.`

Reply:
Thats because you declare it in `createPaymentDateForm()` in the controller. 
Remove it at both places and then register it directly with the DateTimePickerType:
```
            ->add('paymentDate', DateTimePickerType::class, array_merge($dateTimeOptions, [
                 'label' => 'label.paymentDate',
                 'required' => true,
                 'time_increment' => 30,
             ]));
```

Reply:
done, but can you explain to me what `time_increment` does?

Reply:
It defines the minutes in the picker, where you choose the minutes via dropdown. Normally you can choose from 0-59 every single minute, which I believe is not necessary here. You can also delete the option completely, I guess most people will not choose an exact minute there. What do you think?

Reply:
Ok, I guess I can remove it. I modeled the payment_date as `type: 'date' , so it has no time information. Maybe the types  in the entity class are wrong. But I don't know which type can be used instead of `DateTime`. `Date` seems not to exist.